hs_core/templates/search/search.html

```diff
@@ -396,26 +396,63 @@ <h4 class="panel-title">
             });
         }
 
-        var formURL = function(key, value){
+        var formLinkButtonURL = function(key, value){
             var currentURL = window.location.href.split('#')[0];
             var currentHash = window.location.href.split('#')[1];
             var requestURL = currentURL + "&selected_facets="+key+"_exact:"+value + "#" + currentHash;
             map = null;
             window.location = requestURL;
             var checkboxId =  key + "-" + value;
-            localStorage[checkboxId] = true;
+            var localStorageCheckboxId = 'search-' + checkboxId;
+            localStorage[localStorageCheckboxId] = 'true';
+        }
+        
+        var buildURLOnCheckboxes = function () {
+            var requestURL = '';
+            $(".faceted-selections").each(function () {
+                var checkboxId = $(this).attr("id");
+                var localStorageCheckboxId = 'search-' + checkboxId;
+                var val = localStorage[localStorageCheckboxId];
+                var isChecked = val === 'true';
+                if(isChecked){
+                    var arr = $(this).val().split(",");
+                    var key = arr[0];
+                    var value = arr[1];
+                    requestURL += "&selected_facets="+key+"_exact:"+value;
+                }
+            });
+            return requestURL;
+        }
+        var popCheckboxes = function() {
+            $(".faceted-selections").each(function () {
+                var checkboxId = $(this).attr("id");
+                var localStorageCheckboxId = 'search-' + checkboxId;
+                var val = localStorage[localStorageCheckboxId];
+                var isChecked = val !== undefined ? val == 'true' : false;
+                $(this).prop("checked", isChecked);
+            });
+        }
+
+        var clearCheckboxes = function() {
+            $(".faceted-selections").each(function () {
+                var checkboxId = $(this).attr("id");
+                var localStorageCheckboxId = 'search-' + checkboxId;
+                localStorage[localStorageCheckboxId] = 'false';
+                localStorage.removeItem(localStorageCheckboxId);
```

Comment:
@xuezhaokun If the user leaves the page this will leave the checked boxes in localStorage, and when they come back they'll still be stored and, if their IDs have changed, they'll be left forever and possibly cause problems later.

Can you swap out `sessionStorage` for `localStorage`? It works the same way, but is not stored once the browser session on the site has ended. Just a little safer that way.

If you make that change, then +1 to this PR.